Source/SwiftLintFramework/Rules/Idiomatic/NoConstantConditionRule.swift

```diff
@@ -0,0 +1,566 @@
+            Example("let x = foo() ? \"a\" : \"b\""),
+            Example("guard foo() {} else {}"),
+            Example("switch foo() { default: break }")
+        ],
+        triggeringExamples: [
+            Example("if ↓true {}"),
+            Example("if ↓!true {}"),
+            Example("if ↓false {}"),
+            Example("if ↓!false {}"),
+            Example("if ↓!((false)) {}"),
+            Example("if ↓!(!(false)) {}"),
+            Example("if ↓!(false) {}"),
+            Example("if ↓0 < 1 {}"),
+            Example("if ↓0.1 < 1.1 {}"),
+            Example("if ↓\"a\" < \"b\" {}"),
+            Example("if ↓true == false {}"),
+            Example("if ↓nil != nil {}"),
+            Example("if ↓1 < 2 && 2 < 3 {}"),
+            Example("if ↓0...10 ~= 5 {}"),
+            Example("if ↓!(0 < 1) {}"),
+            Example("if ↓1 < 0 {}"),
+            Example("if ↓1+1 < 1+2 {}"),
+            Example("if ↓1+1 > 1+2 {}"),
+            Example("if ↓1+1 == 1+2 {}"),
+            Example("if ↓1+1 == 1+2 && true {}"),
+            Example("if ↓1+1 == 1+2 && false {}"),
+            Example("if ↓1 < 2 && (2 < 3 || 3 < 4) {}"),
+            Example("if ↓\"foo\" == \"foo\" {}"),
+            Example("if ↓\"foo\" != \"foo\" {}"),
+            Example("if ↓true, false {}"),
+            Example("if ↓true, true {}"),
+            Example("if ↓foo(), false {}"),
+            Example("if ↓foo < 2, false {}"),
+            Example("if ↓foo(), 1 > 2 {}"),
+            Example("if ↓1 < 2, true {}"),
+            Example("if ↓1 < 2, false {}"),
+            Example("if ↓foo() && false {}"),
+            Example("if ↓foo() && !true {}"),
+            Example("if ↓foo() && !(true) {}"),
+            Example("if ↓foo() && !(!false) {}"),
+            Example("if ↓foo() && -5 == -4 {}"),
+            Example("if ↓foo() && -5 > -4 {}"),
+            Example("if ↓foo() && (2+1 != 4-1) {}"),
+            Example("if ↓foo() < bar && (x == nil && (!a && false)) {}"),
+            Example("if ↓foo() < bar && (true == false) {}"),
+            Example("if ↓!(foo() && false) {}"),
+            Example("if ↓!!(foo() && false) {}"),
+            Example("if ↓!!!(foo() && false) {}"),
+            Example("if ↓foo(), false {}"),
+            Example("if foo() {} else if ↓false {}"),
+            Example("let x = ↓true ? \"a\" : \"b\""),
+            Example("let x = ↓(foo() && false) ? \"a\" : \"b\""),
+            Example("let x = (↓(foo() && false) ? \"a\" : \"b\")"),
+            Example("guard ↓true {} else {}"),
+            Example("guard ↓foo() && false {} else {}"),
+            Example("guard ↓2 < 3 {} else {}"),
+            Example("switch ↓2 < 3 { default: break }"),
+            Example("switch ↓\"foo\" { default: break }"),
+            Example("switch ↓foo() && false { default: break }")
+        ]
+    )
+
+    func preprocess(file: SwiftLintFile) -> SourceFileSyntax? {
+        return file.foldedSyntaxTree
```

Comment:
returns not required in one liners

Reply:
I know, but personally I prefer being explicit. I think it improves readability because the person reading to code can immediately see the `return` statement and doesn’t have to count the amount of lines. That’s being said, if this is a requirement that this project wants to enforce I’m happy to change it (sounds like there should be linting rule for that 😁).

Reply:
https://realm.github.io/SwiftLint/implicit_return.html will detect that, but it's disabled in SwiftLint's `.swiftlint.yml` :-)

I don't think there's any policy here, as I can see plenty of examples, elsewhere in the code, so I think you're fine to keep it if you prefer.